### ISSUE TITLE
IDE0060 shows a warning for extension methods

### ISSUE THREAD

**Issue by ArneKlein** [dotnet/roslyn#36085]
**VS 2019 16.1.1**: 

**Steps to Reproduce**:

1. Create an extension method, e.g.

```csharp
using System;
using System.Web.Mvc;

namespace MvcApplication1.Helpers
{
     public static class LabelExtensions
     {
          public static string Label(this HtmlHelper helper, string target, string text)
          {
               return String.Format("<label for='{0}'>{1}</label>", target, text);
          }
     }
}
```

2. A warning is shown, that the helper parameter is not used, which is correct but IMHO not the expected behaviour for an extension method.

**Expected Behavior**:
No warning is shown for a parameter passed with the this flag.
**Actual Behavior**:
A warning is shown

**Comment by CyrusNajmabadi** [comment]
you can name the variable `this HtmlHelper _` to disable the warning.

**Comment by ArneKlein** [comment]
Thank you for your quick response, that helps with the warning.

I would like it better though, if parameters marked with this are excluded from this rule if possible. 

If that won't work for any reason, a hint for the naming convention in the warning, or even better a quick fix suggesting renaming the parameter would be great.

**Comment by CyrusNajmabadi** [comment]
> I would like it better though, if parameters marked with `this` are excluded from this rule if possible.

I disagree.  I would *definitely* want to know if i had an extension method that didn't actually use the instance that it was extending.  To me, that seems like a big smell and something to draw attention to.

**Comment by mavasani** [comment]
I agree with @CyrusNajmabadi - this can uncover accidental functional issues, and the method should likely be converted to a regular static method without the "this" parameter.

**Comment by CyrusNajmabadi** [comment]
Closing out.  This is the intended design.
